tools/alara_output_processing/alara_output_plotting.py

```diff
@@ -0,0 +1,333 @@
+    adf,
+    run_lbl,
+    variable,
+    nuclides = None,
+    time_unit='s',
+    sort_by_time='shutdown',
+    head=None
+):
+    '''
+    Prepare an ALARADFrame containing data from multiple runs and potentially
+        multiple responses for plotting by filtering rows to a specified run,
+        variable, and potentially nuclides. After filtering, a pivot table is
+        created indexed by nuclide with values for each cooling time.
+
+    Arguments:
+        adf (alara_output_processing.ALARADFrame): ALARADFrame containing 
+            response data from one or more ALARA runs.
+        run_lbl (str): Distinguisher of the specified ALARA run.
+        variable (str): Name of the response variable.
+        nuclides (str, list, or None, optional): Optional parameter for
+            nuclide selection. For a single nuclide, input should be a string
+            of the form "element-A". For multiple individual nuclides, they
+            should be input as a list. To filter all nuclides of a given
+            element, provide the chemical symbol of the desired element,
+            either as a stand-alone string or in a list with other elements or
+            nuclides. To include the "total" row produced by ALARA, write
+            "total". If no nuclides or elements provided, the filtering will
+            pass through all nuclides matching the run_lbl and variable
+            parameters.
+            (Defaults to None)
+        time_unit (str, optional): Optional paramter to set units for cooling
+            times. Accepted values: 's', 'm', 'h', 'd', 'w', 'y', 'c'.
+            (Defaults to 's')
+        sort_by_time (str, optional): Option to sort the ALARADFrame by the
+            data in a particular time column.
+            (Defaults to 'shutdown')
+        head (int or None, optional): Option by which to truncate the
+            ALARADFrame to a particular number of rows.
+            (Defaults to None)
+        
+    Returns:
+        times (list of floats): Chronological list of converted cooling times.
+        filtered (alara_output_processing.ALARADFrame): Modified copy of input
+            adf containing only rows that match all conditions in run_lbl,
+            variable, and (if present) nuclides.
+        piv (pandas.DataFrame): Pivot table indexed by nuclides with values
+            for each cooling time.
+    '''
+
+    filter_dict = {
+        'run_lbl' : run_lbl, 'variable' : adf.VARIABLE_ENUM[variable]
+    }
+    if nuclides:
+        filter_dict['nuclide'] = nuclides
+        
+    filtered = adf.filter_rows(filter_dict)
+    
+    piv = filtered.pivot(
+        index='nuclide',
+        columns='time',
+        values='value'
+    )
+    times = aop.convert_times(
+        array(adf['time'].unique().tolist()), 's', time_unit
+    )
+
+    if head:
+        sort_by_time = aop.extract_time_vals([sort_by_time])[0]
+        piv = piv.sort_values(sort_by_time, ascending=False).head(head)
```

Comment:
1. why does this sorting only happen if `head`?
2. the sorting would be a good stand alone function as I anticipate a user wanting to sort a dataframe and although it's only a couple of lines, it's not super obvious how